apps/server/src/modules/oauth-adapter/service/oauth-adapter.service.spec.ts

```diff
@@ -96,64 +95,33 @@ describe('OauthAdapterServive', () => {
 		});
 
 		describe('when no token got returned', () => {
-			const setup = () => {
-				const error = new Error('unknown error');
+			const setup = (errorType: { isAxios: boolean }) => {
+				const error = errorType.isAxios ? new AxiosError() : new Error('unknown error');
 				httpService.post.mockReturnValueOnce(throwError(() => error));
-
-				return {
-					error,
-				};
 			};
-
-			it('should throw an error', async () => {
-				const { error } = setup();
-
-				const resp = service.sendTokenRequest('tokenEndpoint', testPayload);
-
-				await expect(resp).rejects.toEqual(error);
-			});
-		});
-
-		describe('when error got returned', () => {
-			describe('when error is a unknown error', () => {
-				const setup = () => {
-					const error = new Error('unknown error');
-					httpService.post.mockReturnValueOnce(throwError(() => error));
-
-					return {
-						error,
-					};
-				};
-
-				it('should throw the default sso error', async () => {
-					const { error } = setup();
+			describe('and error is not an axios error', () => {
+				it('should throw an OAuthAdapterErrorLoggableException', async () => {
+					setup({ isAxios: false });
 
 					const resp = service.sendTokenRequest('tokenEndpoint', testPayload);
 
-					await expect(resp).rejects.toEqual(error);
+					await expect(resp).rejects.toThrowError(OAuthAdapterErrorLoggableException);
+					await expect(resp).rejects.toMatchObject({
+						errorMessage: expect.any(String),
+						error: expect.any(Error),
+					});
 				});
 			});
-
-			describe('when error is a axios error', () => {
-				const setup = () => {
-					const error = {
-						error: 'invalid_request',
-					};
-					const axiosError: AxiosError = axiosErrorFactory.withError(error).build();
-
-					httpService.post.mockReturnValueOnce(throwError(() => axiosError));
-
-					return {
-						axiosError,
-					};
-				};
-
-				it('should throw an error', async () => {
-					const { axiosError } = setup();
+			describe('and error is an axios error', () => {
```

Comment:
I think we don't need 2 diferent tests here
if the error is axios or other, we don't care at this level
important is that we throw a loggableException